FabricioOjeda Libreria.cpp

```diff
@@ -0,0 +1,387 @@
+        }
+    }
+
+    void buscarPorLibro(int id) {
+        for (int i = 0; i < cantidad; i++) {
+            if (listaKits[i]->contieneLibro(id)) {
+                listaKits[i]->mostrar();
+            }
+        }
+    }
+
+    Kit* getKit(int i) { return listaKits[i]; }
+    int getCantidad() { return cantidad; }
+};
+
+#endif
+
+//----------------------------------------------------------- PEDIDO.H -----------------------------------------------------------
+
+#ifndef PEDIDO_H
+#define PEDIDO_H
+
+#include "ListaKits.h"
+
+class Pedido {
+private:
+    ListaKits* kits;
+    int maxNumeroKits;
+
+public:
+    Pedido(int maxKits = 5) {
+        maxNumeroKits = maxKits;
+        kits = new ListaKits(maxKits);
+    }
+
+    ~Pedido() {
+        delete kits;
+    }
+
+    bool agregarKit(Kit* k) {
+        for (int i = 0; i < kits->getCantidad(); i++) {
+            Kit* existente = kits->getKit(i);
+            if (existente->contieneLibro(k->getLibro1()->getId()) ||
+                existente->contieneLibro(k->getLibro2()->getId())) {
+                cout << "Error: libro repetido en pedido.\n";
+                return false;
+            }
+        }
+        if (kits->getCantidad() < maxNumeroKits) {
+            kits->agregarKit(k);
+            return true;
+        }
+        cout << "Error: pedido lleno.\n";
+        return false;
+    }
+
+    void mostrar() {
+        cout << "Pedido con " << kits->getCantidad() << " kits:" << endl;
+        kits->mostrar();
+    }
+};
+
+#endif
+
+//----------------------------------------------------------- LISTAPEDIDOS.H -----------------------------------------------------------
+
+#ifndef LISTAPEDIDOS_H
+#define LISTAPEDIDOS_H
+
+#include "Pedido.h"
+
+class ListaPedidos {
+private:
+    Pedido** listaPedidos;
+    int capacidad;
+    int cantidad;
+
+public:
+    ListaPedidos(int cap = 50) {
+        capacidad = cap;
+        cantidad = 0;
+        listaPedidos = new Pedido * [capacidad];
+    }
+
+    ~ListaPedidos() {
+        for (int i = 0; i < cantidad; i++) {
+            delete listaPedidos[i];
+        }
+        delete[] listaPedidos;
+    }
+
+    void agregarPedido(Pedido* p) {
+        if (cantidad < capacidad) {
+            listaPedidos[cantidad] = p;
+            cantidad++;
+        }
+    }
+
+    void mostrar() {
+        for (int i = 0; i < cantidad; i++) {
+            listaPedidos[i]->mostrar();
+        }
+    }
+};
+
+#endif
+
+//----------------------------------------------------------- LIBRERIA.H -----------------------------------------------------------
+
+#ifndef LIBRERIA_H
+#define LIBRERIA_H
+
+#include "ListaLibros.h"
+#include "ListaKits.h"
+#include "ListaPedidos.h"
+
+class Libreria {
+private:
+    ListaLibros* listaLibros;
+    ListaKits* listaKits;
+    ListaPedidos* listaPedidos;
+
+public:
+    Libreria() {
+        listaLibros = new ListaLibros();
+        listaKits = new ListaKits();
+        listaPedidos = new ListaPedidos();
+    }
```

Comment:
Utilizar siempre los argumentos
```suggestion
    Libreria(int capacidadLibros, int capacidadKits, int capacidadPedidos) {
        listaLibros = new ListaLibros(capacidadLibros);
        listaKits = new ListaKits(capacidadKits);
        listaPedidos = new ListaPedidos(capacidadPedidos);
    }
```